// undraw-ui/es/components/context-menu/index.js
import { defineComponent as g, useCssVars as y, ref as k, reactive as h, onMounted as C, openBlock as i, createElementBlock as u, Fragment as M, renderList as b, withDirectives as w, createBlock as N, unref as d, createCommentVNode as B, createElementVNode as E, toDisplayString as H, nextTick as U } from "vue";
import { r as V, c as D } from "../../util/dom.js";
import { UIcon as I, vCleanHtml as L } from "undraw-ui";
import "../../util/clean-html.js";
import { _ as R } from "../anchor/index.js";
/*! UndrawUi v1.2.6 */
const W = ["onClick"], X = /* @__PURE__ */ g({
  __name: "context-menu",
  props: {
    data: {},
    positon: {}
  },
  setup(s) {
    const r = s;
    y((e) => ({
      "41a5d300": c.top,
      "19de7888": c.left
    }));
    const o = k(), c = h({ top: "0px", left: "0px" });
    function m() {
      let { x: e, y: t } = r.positon;
      U(() => {
        const { clientHeight: n, clientWidth: a } = document.body, { clientHeight: l, clientWidth: f } = o.value;
        t + l > n && (t = n - l - 5), e + f > a && (e = a - f - 5), c.top = t + "px", c.left = e + "px";
      });
    }
    function p() {
      V(o.value);
    }
    function _(e) {
      let t = r.data[e];
      t.callback && t.callback(p);
    }
    function v(e, t) {
      return e !== t && e && e.contains(t);
    }
    function x(e) {
      var t = e.pageX, n = e.pageY;
      return { x: t, y: n };
    }
    return C(() => {
      m(), document.body.addEventListener("mousedown", (e) => {
        x(e);
        const t = o.value;
        !v(t, e.target) && t != e.target && p();
      });
    }), (e, t) => (i(), u("div", {
      ref_key: "contextMenuRef",
      ref: o,
      class: "u-context-menu"
    }, [
      (i(!0), u(M, null, b(e.data, (n, a) => (i(), u("div", {
        class: "item",
        key: a,
        onClick: (l) => _(a)
      }, [
        n.svg ? w((i(), N(d(I), {
          key: 0,
          style: { "margin-right": "5px" }
        }, null, 512)), [
          [d(L), n.svg]
        ]) : B("", !0),
        E("span", null, H(n.label), 1)
      ], 8, W))), 128))
    ], 512));
  }
});
const Y = /* @__PURE__ */ R(X, [["__scopeId", "data-v-8f6f12ad"]]);
function j(s, r) {
  let o = { x: s.pageX, y: s.pageY };
  r.positon = o;
  const { vnode: c } = D(Y, r);
  return c;
}
export {
  j as U
};
