// undraw-ui/es/components/crud/index.js
import { defineComponent as O, ref as m, reactive as x, provide as H, onMounted as J, openBlock as v, createElementBlock as K, withDirectives as Q, createElementVNode as a, renderSlot as b, vShow as W, createBlock as w, unref as r, withCtx as l, createTextVNode as y, createCommentVNode as p, createVNode as d, createSlots as M, renderList as S, normalizeProps as U, guardReactiveProps as X, mergeProps as Y, nextTick as L, pushScopeId as Z, popScopeId as ee } from "vue";
import { debounce as te, throttle as R, UTable as oe, UDialog as le, UGroup as re } from "undraw-ui";
import { ElMessageBox as ae, ElButton as u, ElTooltip as T } from "element-plus";
import { c as se } from "../../util/clone-deep.js";
import "../../util/clean-html.js";
import { _ as ne } from "../anchor/index.js";
/*! UndrawUi v1.2.6 */
const C = (g) => (Z("data-v-155b2cac"), g = g(), ee(), g), ie = { class: "u-crud" }, de = /* @__PURE__ */ C(() => /* @__PURE__ */ a("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 1024 1024"
}, [
  /* @__PURE__ */ a("path", {
    fill: "currentColor",
    d: "M480 480V128a32 32 0 0 1 64 0v352h352a32 32 0 1 1 0 64H544v352a32 32 0 1 1-64 0V544H128a32 32 0 0 1 0-64h352z"
  })
], -1)), ue = /* @__PURE__ */ C(() => /* @__PURE__ */ a("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 1024 1024"
}, [
  /* @__PURE__ */ a("path", {
    fill: "currentColor",
    d: "m199.04 672.64 193.984 112 224-387.968-193.92-112-224 388.032zm-23.872 60.16 32.896 148.288 144.896-45.696L175.168 732.8zM455.04 229.248l193.92 112 56.704-98.112-193.984-112-56.64 98.112zM104.32 708.8l384-665.024 304.768 175.936L409.152 884.8h.064l-248.448 78.336L104.32 708.8zm384 254.272v-64h448v64h-448z"
  })
], -1)), ce = /* @__PURE__ */ C(() => /* @__PURE__ */ a("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 1024 1024"
}, [
  /* @__PURE__ */ a("path", {
    fill: "currentColor",
    d: "M160 256H96a32 32 0 0 1 0-64h256V95.936a32 32 0 0 1 32-32h256a32 32 0 0 1 32 32V192h256a32 32 0 1 1 0 64h-64v672a32 32 0 0 1-32 32H192a32 32 0 0 1-32-32V256zm448-64v-64H416v64h192zM224 896h576V256H224v640zm192-128a32 32 0 0 1-32-32V416a32 32 0 0 1 64 0v320a32 32 0 0 1-32 32zm192 0a32 32 0 0 1-32-32V416a32 32 0 0 1 64 0v320a32 32 0 0 1-32 32z"
  })
], -1)), fe = /* @__PURE__ */ C(() => /* @__PURE__ */ a("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 1024 1024"
}, [
  /* @__PURE__ */ a("path", {
    fill: "currentColor",
    d: "m795.904 750.72 124.992 124.928a32 32 0 0 1-45.248 45.248L750.656 795.904a416 416 0 1 1 45.248-45.248zM480 832a352 352 0 1 0 0-704 352 352 0 0 0 0 704z"
  })
], -1)), me = /* @__PURE__ */ C(() => /* @__PURE__ */ a("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 1024 1024"
}, [
  /* @__PURE__ */ a("path", {
    fill: "currentColor",
    d: "M771.776 794.88A384 384 0 0 1 128 512h64a320 320 0 0 0 555.712 216.448H654.72a32 32 0 1 1 0-64h149.056a32 32 0 0 1 32 32v148.928a32 32 0 1 1-64 0v-50.56zM276.288 295.616h92.992a32 32 0 0 1 0 64H220.16a32 32 0 0 1-32-32V178.56a32 32 0 0 1 64 0v50.56A384 384 0 0 1 896.128 512h-64a320 320 0 0 0-555.776-216.384z"
  })
], -1)), ve = { style: { width: "100%", display: "flex", "justify-content": "flex-end" } }, pe = /* @__PURE__ */ O({
  name: "UCrud",
  __name: "crud",
  props: {
    table: {},
    crud: { default: () => ({}) },
    form: {},
    maxHeight: {}
  },
  emits: ["inputBlur", "sortChange"],
  setup(g, { expose: E, emit: I }) {
    const n = g, k = m(!0), N = m(!1), s = x({
      selections: [],
      btnLoading: !1,
      title: ""
    }), c = x({
      form: !1
    }), _ = m(), z = m();
    function D() {
      let e = n.crud;
      c.form = !0, L(() => {
        s.title = "新增", z.value.resetFields(), e.beforeSave && e.beforeSave();
      });
    }
    const $ = (e) => {
      let t = n.crud;
      s.title = "修改", c.form = !0, L(() => {
        z.value.resetFields(), n.form.data = se(e), t.beforeUpdate && t.beforeUpdate();
      });
    };
    function P(e) {
      $(e);
    }
    H("edit", P);
    const A = te((e) => {
      ae.confirm("你确定要删除吗？", "系统提示", {
        confirmButtonText: "确定",
        cancelButtonText: "取消",
        type: "warning",
        draggable: !0
      }).then(() => {
        n.crud.remove && n.crud.remove(e, f);
      });
    }), F = R(() => {
      z.value.validate((e) => {
        var t, h;
        if (e) {
          let i = n.crud;
          s.title == "新增" ? i.save && i.save((t = n.form) == null ? void 0 : t.data, f) : s.title == "修改" && i.update && i.update((h = n.form) == null ? void 0 : h.data, f);
        }
      });
    }), f = R(() => {
      var e;
      c.form = !1, (e = _.value) == null || e.reload();
    });
    function j(e) {
      n.table.sort == "custom" && f(), I("sortChange", e);
    }
    const B = m(), q = m();
    return H("u-crud", n.crud), J(() => {
      q.value = B.value.clientHeight;
    }), E({
      refresh: f
    }), (e, t) => {
      var h, i;
      return v(), K("div", ie, [
        Q(a("div", null, [
          b(e.$slots, "query", {}, void 0, !0)
        ], 512), [
          [W, k.value]
        ]),
        a("div", {
          ref_key: "toolRef",
          ref: B,
          class: "tool"
        }, [
          a("div", null, [
            e.crud.save ? (v(), w(r(u), {
              key: 0,
              class: "u-small",
              type: "primary",
              onClick: D
            }, {
              icon: l(() => [
                de
              ]),
              default: l(() => [
                y(" 新增 ")
              ]),
              _: 1
            })) : p("", !0),
            e.crud.update ? (v(), w(r(u), {
              key: 1,
              class: "u-small",
              disabled: ((h = s.selections) == null ? void 0 : h.length) != 1,
              type: "success",
              onClick: t[0] || (t[0] = (o) => $(s.selections[0]))
            }, {
              icon: l(() => [
                ue
              ]),
              default: l(() => [
                y(" 修改 ")
              ]),
              _: 1
            }, 8, ["disabled"])) : p("", !0),
            e.crud.remove ? (v(), w(r(u), {
              key: 2,
              class: "u-small",
              loading: N.value,
              disabled: ((i = s.selections) == null ? void 0 : i.length) == 0,
              type: "danger",
              onClick: t[1] || (t[1] = (o) => r(A)(s.selections))
            }, {
              icon: l(() => [
                ce
              ]),
              default: l(() => [
                y(" 删除 ")
              ]),
              _: 1
            }, 8, ["loading", "disabled"])) : p("", !0),
            b(e.$slots, "tool", {}, void 0, !0)
          ]),
          a("div", null, [
            e.$slots.search ? (v(), w(r(T), {
              key: 0,
              content: k.value ? "隐藏搜索" : "显示搜索",
              placement: "top"
            }, {
              default: l(() => [
                d(r(u), {
                  size: "small",
                  circle: "",
                  onClick: t[2] || (t[2] = (o) => k.value = !k.value)
                }, {
                  icon: l(() => [
                    fe
                  ]),
                  _: 1
                })
              ]),
              _: 1
            }, 8, ["content"])) : p("", !0),
            d(r(T), {
              content: "刷新",
              placement: "top"
            }, {
              default: l(() => [
                d(r(u), {
                  size: "small",
                  circle: "",
                  onClick: r(f)
                }, {
                  icon: l(() => [
                    me
                  ]),
                  _: 1
                }, 8, ["onClick"])
              ]),
              _: 1
            })
          ])
        ], 512),
        d(r(oe), {
          ref_key: "tableRef",
          ref: _,
          table: e.table,
          "max-height": e.maxHeight,
          onOnSelection: t[3] || (t[3] = (o) => s.selections = o),
          onInputBlur: t[4] || (t[4] = (o) => e.$emit("inputBlur", o)),
          onSortChange: j
        }, M({ _: 2 }, [
          S(e.table.columns, (o, G) => ({
            name: `table-${o.prop}`,
            fn: l((V) => [
              b(e.$slots, `table-${o.prop}`, U(X(V)), void 0, !0)
            ])
          }))
        ]), 1032, ["table", "max-height"]),
        d(r(le), {
          modelValue: c.form,
          "onUpdate:modelValue": t[7] || (t[7] = (o) => c.form = o),
          title: s.title
        }, {
          footer: l(() => [
            a("div", ve, [
              d(r(u), {
                onClick: t[5] || (t[5] = (o) => c.form = !1)
              }, {
                default: l(() => [
                  y("取消")
                ]),
                _: 1
              }),
              d(r(u), {
                type: "primary",
                loading: s.btnLoading,
                onClick: t[6] || (t[6] = (o) => r(F)())
              }, {
                default: l(() => [
                  y("确定")
                ]),
                _: 1
              }, 8, ["loading"])
            ])
          ]),
          default: l(() => [
            e.form ? (v(), w(r(re), {
              key: 0,
              ref_key: "groupRef",
              ref: z,
              form: e.form
            }, M({ _: 2 }, [
              S(e.form.items, (o, G) => ({
                name: `form-${o.prop}`,
                fn: l((V) => [
                  o.component ? p("", !0) : b(e.$slots, `form-${o.prop}`, U(Y({ key: 0 }, V)), void 0, !0)
                ])
              }))
            ]), 1032, ["form"])) : p("", !0)
          ]),
          _: 3
        }, 8, ["modelValue", "title"]),
        b(e.$slots, "default", {}, void 0, !0)
      ]);
    };
  }
});
const ge = /* @__PURE__ */ ne(pe, [["__scopeId", "data-v-155b2cac"]]), ze = ge;
export {
  ze as U
};
