// undraw-ui/es/components/anchor/index.js
import { defineComponent as b, ref as i, onMounted as T, onUnmounted as g, openBlock as d, createElementBlock as v, createElementVNode as p, createVNode as y, unref as E, Fragment as S, renderList as O, normalizeClass as w, toDisplayString as x, pushScopeId as U, popScopeId as k } from "vue";
import { UDivider as A } from "undraw-ui";
/*! UndrawUi v1.2.6 */
const C = (c) => (U("data-v-a0ba3b29"), c = c(), k(), c), I = { class: "u-anchor" }, q = { class: "toc-content" }, H = /* @__PURE__ */ C(() => /* @__PURE__ */ p("h3", { class: "toc-content-heading" }, "目录", -1)), L = { class: "toc-items" }, N = ["onClick"], V = /* @__PURE__ */ b({
  name: "UAnchor",
  __name: "anchor",
  props: {
    container: {},
    scroll: {},
    targetOffset: { default: 0 }
  },
  setup(c) {
    const o = c, l = i(0), s = i({}), t = i({}), h = (e) => {
      switch (e) {
        case "H1":
        case "H2":
          return "d2";
        case "H3":
          return "d3";
        default:
          return "d4";
      }
    }, _ = () => {
      const e = [];
      s.value.forEach((n) => {
        e.push(n.offsetTop);
      });
      const a = (t.value instanceof Element ? t.value.scrollTop : void 0) || document.documentElement.scrollTop || document.body.scrollTop;
      e.forEach((n, f) => {
        a >= n - 10 - o.targetOffset && (l.value = f);
      });
    }, m = (e) => {
      const r = s.value.item(e);
      console.log(r), o.scroll ? t.value.scrollTo({
        top: r.offsetTop - o.targetOffset,
        behavior: "smooth"
      }) : document.documentElement.scrollTo({
        top: r.offsetTop - o.targetOffset,
        behavior: "smooth"
      });
    };
    let u;
    return T(() => {
      let e = document.querySelector(o.container);
      u = new ResizeObserver((r) => {
        o.scroll ? t.value = document.querySelector(o.scroll) : t.value = window, s.value = e.querySelectorAll("h1, h2, h3, h4, h5, h6"), t.value.addEventListener("scroll", _);
      }), u.observe(e);
    }), g(() => {
      t.value.removeEventListener("scroll", _), u.disconnect();
    }), (e, r) => (d(), v("div", I, [
      p("nav", q, [
        H,
        y(E(A)),
        p("ul", L, [
          (d(!0), v(S, null, O(s.value, (a, n) => (d(), v("li", {
            key: n,
            class: w([{ active: l.value == n }, h(a.nodeName)]),
            onClick: (f) => m(n)
          }, x(a.innerText), 11, N))), 128))
        ])
      ])
    ]));
  }
});
const z = (c, o) => {
  const l = c.__vccOpts || c;
  for (const [s, t] of o)
    l[s] = t;
  return l;
}, B = /* @__PURE__ */ z(V, [["__scopeId", "data-v-a0ba3b29"]]), M = B;
export {
  M as U,
  z as _
};
