// undraw-ui/es/components/comment/index.js
import { defineComponent as O, ref as j, reactive as se, inject as R, withDirectives as ee, openBlock as a, createElementBlock as p, createVNode as b, unref as e, normalizeClass as Z, createElementVNode as l, toDisplayString as C, createCommentVNode as f, createBlock as V, withCtx as S, createTextVNode as te, h as N, pushScopeId as oe, popScopeId as ae, renderSlot as q, computed as ne, toRefs as pe, Fragment as le, renderList as ie, withModifiers as he, nextTick as ge, watch as ye, provide as Y, useSlots as ke, mergeProps as we } from "vue";
import { ElButton as re, ElPopover as xe, ElAvatar as me, ElImage as be, ElPagination as Ce } from "element-plus";
import { UEditor as $e, UEmoji as Ie, translate as E, UToast as Le, mergeObject as ze, UIcon as K, UFold as Se } from "undraw-ui";
import { i as Q, a as Be } from "../../util/isEmpty.js";
import { v as Me, i as Ve, s as A, m as Ee } from "../../util/basic.js";
import { v as de } from "../../util/clean-html.js";
import { _ as G } from "../anchor/index.js";
import { h as He } from "../../util/datetime.js";
import { u as Pe } from "../../hooks/use-emoji/index.js";
import { u as Te } from "../../hooks/use-level/index.js";
/*! UndrawUi v1.2.6 */
const je = (v) => (oe("data-v-8ab40e04"), v = v(), ae(), v), Re = { class: "comment-box" }, Ue = {
  key: 0,
  class: "action-box"
}, Ae = /* @__PURE__ */ je(() => /* @__PURE__ */ l("svg", {
  width: "16",
  height: "16",
  viewBox: "0 0 16 16",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  class: "icon"
}, [
  /* @__PURE__ */ l("path", {
    "data-v-48a7e3c5": "",
    "fill-rule": "evenodd",
    "clip-rule": "evenodd",
    d: "M14 1.3335C14.3514 1.3335 14.6394 1.60546 14.6648 1.95041L14.6666 2.00016V14.0002C14.6666 14.3516 14.3947 14.6396 14.0497 14.665L14 14.6668H1.99998C1.64853 14.6668 1.36059 14.3949 1.33514 14.0499L1.33331 14.0002V2.00016C1.33331 1.64871 1.60527 1.36077 1.95023 1.33532L1.99998 1.3335H14ZM13.3333 2.66618H2.66664V13.3328H13.3333V2.66618ZM11.9219 6.7879C11.9719 6.83791 12 6.90574 12 6.97647V11.7993C12 11.9098 11.9104 11.9993 11.8 11.9993H6.81615C6.7975 11.9993 6.77945 11.9968 6.76232 11.992L3.91042 11.9847C3.79996 11.9844 3.71063 11.8947 3.7109 11.7842C3.71102 11.7313 3.73209 11.6807 3.76948 11.6433L6.52468 8.88807C6.62882 8.78393 6.79766 8.78393 6.9018 8.88807L8.17297 10.1593L11.5447 6.7879C11.6489 6.68376 11.8177 6.68376 11.9219 6.7879ZM5.99997 3.99951V5.99951H3.99997V3.99951H5.99997Z"
  })
], -1)), Fe = { class: "btn-box" }, Ne = "div[id^='el-popper-container']", Oe = /* @__PURE__ */ O({
  __name: "input-box",
  props: {
    placeholder: {},
    contentBtn: {},
    parentId: {},
    reply: {},
    cancelBtn: {}
  },
  emits: ["hide", "close"],
  setup(v, { expose: d, emit: n }) {
    const c = v, k = j(""), u = j(!1), i = j(!0), _ = j(), B = j(), L = j([]), z = se({
      imgLength: 0
    }), o = ($) => {
      Q(k.value.replace(/&nbsp;|<br>| /g, "")) ? i.value = !0 : i.value = !1;
    }, { upload: r, emoji: t, mention: s } = R("config"), { submit: g, focus: I, cancelFn: w, mentionSearch: T } = R("comment-fun"), H = () => {
      let $ = c.reply && c.parentId != c.reply.id ? `回复 <span style="color: var(--u-color-success-dark-2);">@${c.reply.user.username}:</span> ${k.value}` : k.value;
      Q(L.value) || ($ += `u-img[${L.value}]`), g({
        content: $,
        parentId: Be(c.parentId, null),
        reply: c.reply,
        clear: () => {
          J(), n("close");
        }
      });
    }, U = () => {
      J(), n("close"), w();
    }, J = () => {
      _.value.clear(), i.value = !0, L.value.length = 0;
    };
    function W($) {
      Q(k.value) && !z.imgLength && (u.value = !1, n("hide", $));
    }
    function X() {
      u.value = !0, I();
    }
    d({
      focus: () => {
        var $;
        return ($ = _.value) == null ? void 0 : $.focus();
      },
      changeMentionShow: ($) => {
        var y;
        return (y = _.value) == null ? void 0 : y.changeMentionShow($);
      }
    });
    const m = ($, y) => {
      var D;
      let h = new Array();
      const x = y ? [y] : (D = B.value) == null ? void 0 : D.files;
      if (x)
        for (let F = 0; F < x.length; F++) {
          let _e = x[F].name;
          Ve(_e) ? h.push(x[F]) : Le({ type: "warn", message: "请选择图片类型文件!", duration: 2500 });
        }
      r && r(h, (F) => L.value.push(...F));
    }, P = R("comment-slot"), M = () => N("div", P.func());
    return ($, y) => ee((a(), p("div", Re, [
      b(e($e), {
        ref_key: "editorRef",
        ref: _,
        modelValue: k.value,
        "onUpdate:modelValue": y[0] || (y[0] = (h) => k.value = h),
        mention: e(s),
        class: Z({ "input-active": u.value }),
        placeholder: c.placeholder,
        "min-height": 64,
        "img-list": L.value,
        onClick: y[1] || (y[1] = () => u.value = !0),
        onFocus: X,
        onInput: o,
        onSubmit: H,
        onPaste: m,
        onMentionSearch: e(T)
      }, null, 8, ["modelValue", "mention", "class", "placeholder", "img-list", "onMentionSearch"]),
      u.value ? (a(), p("div", Ue, [
        l("div", null, [
          b(e(Ie), {
            emoji: e(t),
            onAddEmoji: y[2] || (y[2] = (h) => {
              var x;
              return (x = _.value) == null ? void 0 : x.addText(h);
            })
          }, null, 8, ["emoji"])
        ]),
        e(r) ? (a(), p("div", {
          key: 0,
          class: "picture",
          onClick: y[3] || (y[3] = //@ts-ignore
          (...h) => {
            var x, D;
            return ((x = B.value) == null ? void 0 : x.click) && ((D = B.value) == null ? void 0 : D.click(...h));
          })
        }, [
          Ae,
          l("span", null, C(e(E)("comment.upload")), 1),
          l("input", {
            id: "comment-upload",
            ref_key: "inputRef",
            ref: B,
            type: "file",
            multiple: "",
            onChange: m
          }, null, 544)
        ])) : f("", !0),
        e(P).func ? (a(), V(M, { key: 1 })) : f("", !0),
        l("div", Fe, [
          b(e(re), {
            type: "primary",
            disabled: i.value,
            onClick: H
          }, {
            default: S(() => [
              te(C(c.contentBtn), 1)
            ]),
            _: 1
          }, 8, ["disabled"]),
          c.cancelBtn ? (a(), V(e(re), {
            key: 0,
            onClick: U
          }, {
            default: S(() => [
              te(C(c.cancelBtn), 1)
            ]),
            _: 1
          })) : f("", !0)
        ])
      ])) : f("", !0)
    ])), [
      [e(Me), W, Ne]
    ]);
  }
});
const ve = /* @__PURE__ */ G(Oe, [["__scopeId", "data-v-8ab40e04"]]), ue = /* @__PURE__ */ O({
  __name: "user-card",
  props: {
    uid: {}
  },
  setup(v) {
    const d = j({}), { showInfo: n } = R("comment-fun"), c = R("comment-slot"), k = () => N("div", c.card(d.value));
    return (u, i) => e(c).card ? (a(), V(e(xe), {
      key: 0,
      placement: "top",
      width: 300,
      "show-after": 300,
      onBeforeEnter: i[0] || (i[0] = () => e(n)(u.uid, (_) => d.value = _))
    }, {
      reference: S(() => [
        q(u.$slots, "default")
      ]),
      default: S(() => [
        b(k)
      ]),
      _: 3
    })) : q(u.$slots, "default", { key: 1 });
  }
}), ce = (v) => (oe("data-v-adae4a5b"), v = v(), ae(), v), De = { class: "comment-sub" }, Ze = ["href", "target"], qe = { class: "comment-primary" }, Ge = { class: "comment-main" }, Je = { class: "user-info" }, Ke = ["href", "target"], Qe = { class: "username" }, We = {
  class: "name",
  style: { "max-width": "10em" }
}, Xe = {
  key: 0,
  blank: "true",
  class: "rank"
}, Ye = {
  key: 0,
  class: "address",
  style: { color: "#939393", "font-size": "12px" }
}, et = { class: "time" }, tt = { class: "content" }, nt = {
  class: "imgbox",
  style: { display: "flex" }
}, st = { class: "action-box select-none" }, ot = /* @__PURE__ */ ce(() => /* @__PURE__ */ l("svg", {
  t: "1650360973068",
  viewBox: "0 0 1024 1024",
  version: "1.1",
  xmlns: "http://www.w3.org/2000/svg",
  "p-id": "1168",
  width: "200",
  height: "200"
}, [
  /* @__PURE__ */ l("path", {
    d: "M547.968 138.88c6.656-4.672 14.08-6.976 20.48-5.056 6.08 1.792 22.848 10.752 40.192 56.128 8.576 22.4 27.264 81.536-5.632 197.504a45.44 45.44 0 0 0 42.88 57.984l217.6 3.008h0.448a53.12 53.12 0 0 1 20.096 3.328 16.256 16.256 0 0 1 5.568 3.648 14.464 14.464 0 0 1 3.264 6.4c2.176 7.808 4.608 33.984-0.256 77.248-4.672 41.984-15.936 97.408-38.784 162.368-19.136 54.336-43.52 100.48-81.472 161.792a56.384 56.384 0 0 0-1.664 2.496l-0.128 0.128-1.408 2.112a7.872 7.872 0 0 1-1.28 1.472 3.84 3.84 0 0 1-1.28 0.64 20.48 20.48 0 0 1-6.848 0.96H356.032V421.44c19.712-10.624 40.704-24.576 62.592-47.616 25.472-26.88 51.008-64.768 78.208-121.6 5.568-11.584 9.856-24.384 13.632-36.032l3.072-9.856c2.688-8.448 5.184-16.384 8.064-24.32 8.064-22.4 16.128-36.032 26.368-43.136z m120.96 27.968c-20.48-53.44-48-84.736-81.984-94.912-33.6-9.984-61.952 4.16-76.032 14.08-27.584 19.264-41.28 49.6-50.048 74.048-3.392 9.344-6.464 19.2-9.216 27.968l-2.688 8.448a227.84 227.84 0 0 1-10.432 27.904c-25.28 52.928-47.36 84.544-66.752 104.96-18.944 19.968-36.48 30.464-55.168 39.808a45.376 45.376 0 0 0-25.088 40.576l-0.064 480.64c0 24.96 20.224 45.248 45.184 45.248h423.04c21.76 0 38.144-6.912 50.048-16.96a71.808 71.808 0 0 0 14.528-16.896l0.128-0.256 0.128-0.128 0.832-0.96 1.152-1.92c39.424-63.872 66.816-114.688 88.256-175.68a810.24 810.24 0 0 0 42.048-176.64c5.12-45.632 3.776-81.664-1.6-101.376a77.952 77.952 0 0 0-45.568-52.288 116.544 116.544 0 0 0-45.44-8.64l-192.768-2.688c28.096-115.072 10.048-181.568-2.496-214.336z m-604.864 247.04a45.184 45.184 0 0 1 45.12-47.296h67.008c24.96 0 45.184 20.288 45.184 45.248v480.64c0 24.96-20.224 45.12-45.184 45.12H131.84a45.184 45.184 0 0 1-45.12-43.072l-22.656-480.64z",
    "p-id": "1169"
  })
], -1)), at = /* @__PURE__ */ ce(() => /* @__PURE__ */ l("svg", {
  viewBox: "0 0 1024 1024",
  version: "1.1",
  xmlns: "http://www.w3.org/2000/svg",
  "p-id": "1534"
}, [
  /* @__PURE__ */ l("path", {
    d: "M668.928 166.912c-20.48-53.504-47.9744-84.7872-82.0224-94.9248-33.5872-10.0352-61.952 4.096-76.032 13.9776-27.5456 19.3536-41.216 49.664-50.0224 74.1376-3.3792 9.3184-6.4512 19.0976-9.216 27.9552l-2.6624 8.3968a227.8912 227.8912 0 0 1-10.3936 27.9552c-25.344 52.9408-47.4112 84.5312-66.7648 104.96-18.944 19.968-36.4544 30.464-55.1936 39.7824a45.3632 45.3632 0 0 0-25.088 40.5504l-0.1024 480.7168c0 24.9344 20.2752 45.2096 45.2096 45.2096h423.0656c21.7088 0 38.144-6.912 50.0224-16.9984a72.192 72.192 0 0 0 14.4896-16.896l0.2048-0.2048 0.0512-0.1536 0.8192-1.024 1.2288-1.8944c39.424-63.7952 66.7648-114.688 88.2176-175.616 24.4224-69.4784 36.8128-129.6896 42.0352-176.64 5.12-45.6704 3.7888-81.664-1.5872-101.376a77.9776 77.9776 0 0 0-45.568-52.3264 116.5824 116.5824 0 0 0-45.4144-8.6016l-192.8192-2.6624c28.1088-115.0976 10.0864-181.6064-2.4576-214.3744zM64.0512 413.9008a45.2096 45.2096 0 0 1 45.1584-47.36H176.128c24.9344 0 45.2096 20.2752 45.2096 45.2096v480.6144a45.2096 45.2096 0 0 1-45.2096 45.2096h-44.288a45.2096 45.2096 0 0 1-45.1584-43.0592L64 413.952z",
    "p-id": "1535"
  })
], -1)), lt = { key: 2 }, it = ["onClick"], ct = /* @__PURE__ */ ce(() => /* @__PURE__ */ l("svg", {
  viewBox: "0 0 1024 1024",
  version: "1.1",
  xmlns: "http://www.w3.org/2000/svg",
  "p-id": "1320",
  width: "200",
  height: "200"
}, [
  /* @__PURE__ */ l("path", {
    d: "M147.264 647.296V220.928c0-49.536 40.128-89.728 89.6-89.728H793.6c49.536 0 89.728 40.192 89.728 89.728v426.368c0 49.536-40.128 89.728-89.6 89.728h-145.216a47.04 47.04 0 0 0-28.16 9.408l-194.56 145.792a3.392 3.392 0 0 1-5.12-1.984l-26.752-116.672a47.04 47.04 0 0 0-45.824-36.544H236.992a89.728 89.728 0 0 1-89.728-89.728zM236.864 64A156.928 156.928 0 0 0 80 220.928l0.064 426.368a156.928 156.928 0 0 0 156.928 156.928h94.976l23.232 101.312 0.064 0.448a70.592 70.592 0 0 0 109.696 40.832l190.208-142.592H793.6a156.928 156.928 0 0 0 156.928-156.928l-0.064-426.368A156.928 156.928 0 0 0 793.536 64H236.928z m69.44 442.496a65.344 65.344 0 1 0 0-130.752 65.344 65.344 0 0 0 0 130.752z m268.8-65.344a65.344 65.344 0 1 1-130.752 0 65.344 65.344 0 0 1 130.752 0z m138.368 65.344a65.344 65.344 0 1 0 0-130.752 65.344 65.344 0 0 0 0 130.752z",
    "p-id": "1321"
  })
], -1)), rt = { key: 0 }, dt = /* @__PURE__ */ O({
  __name: "content-box",
  props: {
    reply: { type: Boolean },
    data: {},
    id: {}
  },
  setup(v) {
    const d = v, n = se({
      active: !1
    }), c = j(), k = j(), u = j(), i = ne(() => {
      let m = d.data;
      return ze(m.user, {
        level: 1,
        homeLink: null,
        likeIds: []
      }), s(m), m;
    }), _ = R("config"), { user: B, aTarget: L, show: z, emoji: o, relativeTime: r } = pe(_), { like: t, beforeData: s } = R("comment-fun");
    function g() {
      n.active = !n.active, n.active && ge(() => {
        var m;
        (m = c.value) == null || m.focus();
      });
    }
    function I(m) {
      var M;
      const P = m.target;
      (M = k.value) != null && M.contains(P) || (n.active = !1);
    }
    const w = R("comment-slot"), T = () => N("div", w.info(d.data)), H = () => N("div", w.avatar(d.data)), U = () => N("div", w.action(d.data)), J = () => N("div", w.operate(d.data));
    function W(m) {
      const P = /u-img\[.+?\]/g;
      return m = m.replace(P, (M) => (M = M.substring(6, M.length - 1), u.value = M.split(","), "")), m;
    }
    const X = ne(() => {
      u.value = [];
      let m = W(d.data.content);
      return o && (m = Pe(o.value.allEmoji, m)), m;
    });
    return (m, P) => {
      var M, $, y;
      return a(), p("div", {
        class: Z(["comment", { reply: d.reply }])
      }, [
        l("div", De, [
          b(ue, {
            uid: e(A)(i.value.uid)
          }, {
            default: S(() => {
              var h;
              return [
                e(w).avatar ? (a(), V(H, { key: 0 })) : (a(), p("a", {
                  key: 1,
                  href: i.value.user.homeLink,
                  target: e(L),
                  class: Z([{ "pointer-events-none": !((h = e(z)) != null && h.homeLink) }, "no-underline"]),
                  style: { display: "block" }
                }, [
                  b(e(me), {
                    style: { "margin-top": "5px" },
                    size: 40,
                    fit: "cover",
                    src: i.value.user.avatar
                  }, {
                    default: S(() => [
                      l("span", null, C(i.value.user.username), 1)
                    ]),
                    _: 1
                  }, 8, ["src"])
                ], 10, Ze))
              ];
            }),
            _: 1
          }, 8, ["uid"])
        ]),
        l("div", qe, [
          l("div", Ge, [
            l("div", Je, [
              b(ue, {
                uid: e(A)(i.value.uid)
              }, {
                default: S(() => {
                  var h, x;
                  return [
                    l("a", {
                      href: i.value.user.homeLink,
                      target: e(L),
                      class: Z([{ "pointer-events-none": !((h = e(z)) != null && h.homeLink) }, "no-underline"]),
                      style: { display: "block" }
                    }, [
                      l("div", Qe, [
                        l("span", We, C(i.value.user.username), 1),
                        (x = e(z)) != null && x.level ? (a(), p("span", Xe, [
                          ee(b(e(K), { size: "24" }, null, 512), [
                            [e(de), e(Te)(i.value.user.level || 1)]
                          ])
                        ])) : f("", !0)
                      ])
                    ], 10, Ke)
                  ];
                }),
                _: 1
              }, 8, ["uid"]),
              (M = e(z)) != null && M.address ? (a(), p("span", Ye, "  " + C(i.value.address), 1)) : f("", !0),
              e(w).info ? (a(), V(T, { key: 1 })) : f("", !0),
              l("time", et, C(e(r) ? e(He)(i.value.createTime) : i.value.createTime), 1)
            ]),
            l("div", tt, [
              b(e(Se), { unfold: "" }, {
                default: S(() => [
                  ee(l("div", null, null, 512), [
                    [e(de), X.value]
                  ]),
                  l("div", nt, [
                    (a(!0), p(le, null, ie(u.value, (h, x) => (a(), V(e(be), {
                      key: x,
                      "preview-teleported": "",
                      src: h,
                      style: { height: "72px", padding: "8px 4px" },
                      lazy: "",
                      "preview-src-list": u.value,
                      "initial-index": x
                    }, null, 8, ["src", "preview-src-list", "initial-index"]))), 128))
                  ])
                ]),
                _: 1
              })
            ]),
            l("div", st, [
              ($ = e(z)) != null && $.likes ? (a(), p("div", {
                key: 0,
                class: "item",
                onClick: P[0] || (P[0] = (h) => e(t)(e(A)(i.value.id)))
              }, [
                e(B).likeIds && e(B).likeIds.map(String).indexOf(e(A)(i.value.id)) == -1 ? (a(), V(e(K), { key: 0 }, {
                  default: S(() => [
                    ot
                  ]),
                  _: 1
                })) : (a(), V(e(K), {
                  key: 1,
                  class: "liked",
                  color: "#1e80ff"
                }, {
                  default: S(() => [
                    at
                  ]),
                  _: 1
                })),
                i.value.likes != 0 ? (a(), p("span", lt, C(i.value.likes), 1)) : f("", !0)
              ])) : f("", !0),
              (y = e(z)) != null && y.reply ? (a(), p("div", {
                key: 1,
                ref_key: "btnRef",
                ref: k,
                class: Z(["item", { active: n.active }]),
                onClick: he(g, ["stop"])
              }, [
                b(e(K), null, {
                  default: S(() => [
                    ct
                  ]),
                  _: 1
                }),
                l("span", null, C(n.active ? e(E)("comment.cancelReply") : e(E)("comment.reply")), 1)
              ], 10, it)) : f("", !0),
              e(w).action ? (a(), V(U, { key: 2 })) : f("", !0),
              e(w).operate ? (a(), V(J, { key: 3 })) : f("", !0)
            ]),
            n.active ? (a(), p("div", rt, [
              b(ve, {
                ref_key: "commentRef",
                ref: c,
                "parent-id": e(A)(m.id),
                placeholder: `${e(E)("comment.placeholder2")}@${i.value.user.username}...`,
                reply: i.value,
                "content-btn": e(E)("comment.contentBtn2"),
                style: { "margin-top": "12px" },
                onHide: I,
                onClose: P[1] || (P[1] = (h) => n.active = !1)
              }, null, 8, ["parent-id", "placeholder", "reply", "content-btn"])
            ])) : f("", !0)
          ]),
          q(m.$slots, "default", {}, void 0, !0)
        ])
      ], 2);
    };
  }
});
const fe = /* @__PURE__ */ G(dt, [["__scopeId", "data-v-adae4a5b"]]), ut = (v) => (oe("data-v-ac954c19"), v = v(), ae(), v), pt = {
  key: 0,
  class: "reply-box"
}, mt = { class: "reply-list" }, vt = {
  key: 0,
  class: "fetch-more"
}, ft = { key: 0 }, _t = { key: 1 }, ht = { key: 0 }, gt = /* @__PURE__ */ ut(() => /* @__PURE__ */ l("svg", {
  width: "12",
  height: "12",
  viewBox: "0 0 12 12",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg",
  class: ""
}, [
  /* @__PURE__ */ l("path", {
    "data-v-d6f79dbc": "",
    "fill-rule": "evenodd",
    "clip-rule": "evenodd",
    d: "M5.99976 7.93206L10.0656 3.86619C10.1633 3.76856 10.3215 3.76856 10.4192 3.86619L10.7727 4.21975C10.8704 4.31738 10.8704 4.47567 10.7727 4.5733L6.35331 8.99272C6.15805 9.18798 5.84147 9.18798 5.6462 8.99272L1.22679 4.5733C1.12916 4.47567 1.12916 4.31738 1.22679 4.21975L1.58034 3.86619C1.67797 3.76856 1.83626 3.76856 1.93389 3.86619L5.99976 7.93206Z"
  })
], -1)), yt = {
  key: 1,
  class: "fetch-more"
}, kt = /* @__PURE__ */ O({
  __name: "reply-box",
  props: {
    data: {},
    id: {}
  },
  setup(v) {
    const d = v, n = se({
      loading: !1,
      over: !1,
      currentPage: 1,
      pageSize: 5
    }), { comments: c, replyShowSize: k, page: u } = R("config"), { replyPage: i } = R("comment-fun"), _ = ne(() => {
      let t = {
        total: 0,
        length: 0,
        list: []
      };
      if (d.data) {
        let s = d.data.list.length;
        t = {
          total: d.data.total,
          length: s,
          list: d.data.list
        };
      }
      if (!n.over) {
        let s = t.list.slice(0, k);
        t.list = s;
      }
      return u && (t.list = t.list.slice(0, n.pageSize)), t;
    });
    ye(
      () => {
        var t;
        return (t = d.data) == null ? void 0 : t.total;
      },
      (t) => {
        if (t) {
          let s = Math.ceil(t / n.pageSize);
          n.currentPage > s && z(s);
        }
      }
    );
    const B = () => {
      n.over = !0;
    }, L = (t) => {
      c.forEach((s) => {
        s.id == d.id && s.reply && (s.reply = t);
      });
    }, z = (t) => {
      n.currentPage = t, i({ current: t, size: n.pageSize, parentId: d.id, finish: (s) => L(s) });
    }, o = (t) => {
      z(t);
    }, r = (t) => {
      n.pageSize = t, i({ current: t, size: n.pageSize, parentId: d.id, finish: (s) => L(s) });
    };
    return (t, s) => _.value.length > 0 ? (a(), p("div", pt, [
      l("div", mt, [
        (a(!0), p(le, null, ie(_.value.list, (g, I) => (a(), V(fe, {
          id: t.id,
          key: I,
          data: g,
          reply: ""
        }, null, 8, ["id", "data"]))), 128)),
        _.value.length > (e(k) || 3) ? (a(), p("div", vt, [
          n.loading ? (a(), p("span", ft, C(e(E)("comment.more.loading")), 1)) : (a(), p("div", _t, [
            n.over ? f("", !0) : (a(), p("div", ht, [
              l("span", {
                class: "fetch-more-comment select-none",
                onClick: B
              }, [
                te(C(e(E)("comment.more.prefixTotal")) + C(_.value.total) + C(e(E)("comment.more.suffixTotal")) + " ", 1),
                gt
              ])
            ]))
          ]))
        ])) : f("", !0),
        n.over && e(u) ? (a(), p("div", yt, [
          b(e(Ce), {
            small: "",
            "hide-on-single-page": "",
            layout: "total, prev, pager, next",
            total: _.value.total,
            "current-page": n.currentPage,
            "page-size": n.pageSize,
            onCurrentChange: o,
            onSizeChange: r
          }, null, 8, ["total", "current-page", "page-size"])
        ])) : f("", !0)
      ])
    ])) : f("", !0);
  }
});
const wt = /* @__PURE__ */ G(kt, [["__scopeId", "data-v-ac954c19"]]), xt = {
  key: 0,
  class: "comment-list"
}, bt = /* @__PURE__ */ O({
  __name: "comment-list",
  props: {
    data: {}
  },
  setup(v) {
    return (d, n) => d.data ? (a(), p("div", xt, [
      (a(!0), p(le, null, ie(d.data, (c) => (a(), V(fe, {
        id: e(A)(c.id),
        key: e(A)(c.id),
        data: c
      }, {
        default: S(() => [
          b(wt, {
            id: e(A)(c.id),
            data: c.reply
          }, null, 8, ["id", "data"])
        ]),
        _: 2
      }, 1032, ["id", "data"]))), 128))
    ])) : f("", !0);
  }
});
const Ct = /* @__PURE__ */ G(bt, [["__scopeId", "data-v-33d8416e"]]), $t = { class: "u-comment" }, It = {
  key: 0,
  class: "comment-form"
}, Lt = { class: "header" }, zt = { class: "header-title" }, St = { class: "content" }, Bt = { class: "avatar-box" }, Mt = {
  key: 1,
  class: "comment-list-wrapper"
}, Vt = { class: "title" }, Et = /* @__PURE__ */ O({
  name: "UComment",
  __name: "comment",
  props: {
    config: {}
  },
  emits: ["submit", "like", "replyPage", "showInfo", "focus", "cancel", "getMentionList", "mentionSearch", "before-data"],
  setup(v, { expose: d, emit: n }) {
    const c = v;
    function k() {
      let o = {
        form: !0,
        content: !0,
        level: !0,
        likes: !0,
        address: !0,
        homeLink: !0,
        reply: !0
      };
      Ee(c.config, {
        show: o,
        aTarget: "_blank",
        replyShowSize: 4
      });
    }
    k();
    const { comments: u, show: i } = pe(c.config), _ = ({ content: o, parentId: r, reply: t, clear: s }) => {
      const g = (T) => {
        if (s(), T)
          if (r) {
            let H = u.value.find((U) => U.id == r);
            if (H) {
              let U = H.reply;
              U ? (U.list.unshift(T), U.total++) : H.reply = {
                total: 1,
                list: [T]
              };
            }
          } else
            u.value.unshift(T);
      };
      let w = (o.match(/<span [^>]*data-id="([^"]*)"[^>]*>/g) || []).map((T) => {
        let H = T.match(/data-id="([^"]*)"/);
        return H ? H[1] : null;
      });
      n("submit", { content: o, parentId: r, finish: g, reply: t, mentionList: w });
    }, B = (o, r) => {
      let t = null;
      u.value.forEach((s) => {
        var g;
        if (s.id == o)
          t = s;
        else {
          let I = (g = s.reply) == null ? void 0 : g.list.find((w) => w.id == o);
          I && (t = I);
        }
      }), t && !Q(t.likes) && (t.likes += r);
    }, L = (o) => {
      const r = c.config.user.likeIds;
      r && n("like", o, () => {
        if (r.findIndex((t) => t == o) == -1)
          r.push(o), B(o, 1);
        else {
          let t = r.findIndex((s) => s == o);
          t != -1 && (r.splice(t, 1), B(o, -1));
        }
      });
    }, z = (o) => {
      const { parentId: r, id: t } = o;
      if (r) {
        let s = u.value.find((I) => I.id == r), g = s == null ? void 0 : s.reply;
        if (g) {
          let I = g.list.findIndex((w) => w.id == t);
          I != -1 && (g.list.splice(I, 1), g.total--);
        }
      } else {
        let s = u.value.findIndex((g) => g.id == t);
        s != -1 && u.value.splice(s, 1);
      }
    };
    return Y("config", c.config), Y("comment-fun", {
      like: L,
      showInfo: (o, r) => n("showInfo", o, r),
      replyPage: ({ current: o, size: r, parentId: t, finish: s }) => {
        n("replyPage", { current: o, size: r, parentId: t, finish: s });
      },
      submit: _,
      focus: () => n("focus"),
      cancelFn: () => n("cancel"),
      mentionSearch: (o) => n("mentionSearch", o),
      beforeData: (o) => n("before-data", o)
    }), Y("comment-slot", ke()), d({
      submit: _,
      remove: z
    }), (o, r) => {
      var t, s;
      return a(), p("div", $t, [
        (t = e(i)) != null && t.form ? (a(), p("div", It, [
          q(o.$slots, "header", {}, () => [
            l("div", Lt, [
              l("span", zt, C(e(E)("comment.headerTitle")), 1)
            ])
          ], !0),
          l("div", St, [
            l("div", Bt, [
              b(e(me), {
                size: 40,
                src: o.config.user.avatar
              }, {
                default: S(() => [
                  l("span", null, C(o.config.user.username), 1)
                ]),
                _: 1
              }, 8, ["src"])
            ]),
            b(ve, we(o.$attrs, {
              ref: "inputBox",
              placeholder: e(E)("comment.placeholder"),
              "content-btn": e(E)("comment.contentBtn")
            }), null, 16, ["placeholder", "content-btn"])
          ])
        ])) : f("", !0),
        (s = e(i)) != null && s.content ? (a(), p("div", Mt, [
          q(o.$slots, "default", {}, () => [
            l("div", Vt, C(e(E)("comment.title")), 1)
          ], !0),
          b(Ct, { data: e(u) }, null, 8, ["data"])
        ])) : f("", !0)
      ]);
    };
  }
});
const Ht = /* @__PURE__ */ G(Et, [["__scopeId", "data-v-df3e5cf8"]]), Zt = Ht;
export {
  Zt as U
};
