// undraw-ui/es/components/fold/index.js
import { defineComponent as V, useCssVars as g, computed as N, ref as s, onMounted as U, onUnmounted as $, openBlock as u, createElementBlock as v, createElementVNode as f, normalizeClass as c, createBlock as E, unref as t, withCtx as p, createTextVNode as m, toDisplayString as _, createCommentVNode as y, renderSlot as x, createVNode as w } from "vue";
import { translate as d } from "undraw-ui";
import { ElButton as k } from "element-plus";
import { _ as z } from "../anchor/index.js";
/*! UndrawUi v1.2.6 */
const F = { class: "u-fold" }, H = { class: "action-box select-none" }, M = /* @__PURE__ */ V({
  name: "UFold",
  __name: "fold",
  props: {
    line: { default: 5 },
    unfold: { type: Boolean },
    position: { default: "left" }
  },
  setup(B) {
    const C = B;
    g((e) => ({
      "84a88e26": b.value
    }));
    const b = N(() => {
      let e = Math.trunc(Number(C.line));
      return e > 0 ? e : 1;
    }), o = s(!0), r = s(!1), n = s(), i = s();
    let l;
    return U(() => {
      l = new ResizeObserver((e) => {
        o.value && n.value && i.value && (r.value = i.value.clientHeight < n.value.scrollHeight);
      }), l.observe(n.value);
    }), $(() => {
      l == null || l.disconnect();
    }), (e, a) => (u(), v("div", F, [
      f("div", {
        ref_key: "textBox",
        ref: i,
        class: c(["txt-box", { "over-hidden": o.value }])
      }, [
        f("div", {
          ref_key: "divBox",
          ref: n
        }, [
          r.value && e.unfold && e.position == "right" ? (u(), E(t(k), {
            key: 0,
            onClick: a[0] || (a[0] = (h) => o.value = !o.value),
            class: c({ "over-hidden": o.value, "end-btn": 1 }),
            type: "primary",
            plain: "",
            link: ""
          }, {
            default: p(() => [
              m(_(o.value ? t(d)("fold.unfold") : t(d)("fold.fold")), 1)
            ]),
            _: 1
          }, 8, ["class"])) : y("", !0),
          x(e.$slots, "default", {}, void 0, !0)
        ], 512)
      ], 2),
      f("div", H, [
        r.value && e.unfold && e.position == "left" ? (u(), v("div", {
          key: 0,
          class: "expand-btn",
          onClick: a[1] || (a[1] = (h) => o.value = !o.value)
        }, [
          x(e.$slots, "expand", { fold: o.value }, () => [
            w(t(k), {
              type: "primary",
              plain: "",
              link: ""
            }, {
              default: p(() => [
                m(_(o.value ? t(d)("fold.unfold") : t(d)("fold.fold")), 1)
              ]),
              _: 1
            })
          ], !0)
        ])) : y("", !0)
      ])
    ]));
  }
});
const O = /* @__PURE__ */ z(M, [["__scopeId", "data-v-da19aa42"]]), T = O;
export {
  T as U
};
