// undraw-ui/es/components/form/index.js
import { defineComponent as $, openBlock as m, createElementBlock as f, Fragment as b, createBlock as i, unref as n, mergeProps as s, createCommentVNode as p, withCtx as h, renderList as S, createElementVNode as w, toDisplayString as V, renderSlot as v, ref as D, createVNode as g, withDirectives as I, normalizeClass as R, normalizeStyle as z, createSlots as B, normalizeProps as C, guardReactiveProps as Y, vShow as N } from "vue";
import { ElInput as T, ElSelect as W, ElOption as j, ElDatePicker as q, ElForm as M, ElFormItem as O } from "element-plus";
import { t as u, m as L } from "../../util/basic.js";
import "../../util/clean-html.js";
import { T as k } from "../../util/datetime.js";
import { USearch2 as A } from "undraw-ui";
import { _ as E } from "../anchor/index.js";
/*! UndrawUi v1.2.6 */
const G = {
  key: 0,
  style: { display: "flex", "justify-content": "space-between" }
}, H = { style: { color: "var(--el-text-color-secondary)", "font-size": "13px" } }, J = /* @__PURE__ */ $({
  __name: "item",
  props: {
    item: {},
    data: {}
  },
  setup(c) {
    const y = [
      {
        text: "今天",
        value: /* @__PURE__ */ new Date()
      },
      {
        text: "昨天",
        value: new k().add(-1, "day")
      },
      {
        text: "一周前",
        value: new k().add(-1, "week")
      }
    ];
    return (e, r) => e.item.prop && e.item.component ? (m(), f(b, { key: 0 }, [
      e.item.component.name == "el-input" ? (m(), i(n(T), s({
        key: 0,
        modelValue: e.data[e.item.prop],
        "onUpdate:modelValue": r[0] || (r[0] = (l) => e.data[e.item.prop] = l),
        clearable: "",
        style: { width: n(u)(e.item.component.width) },
        placeholder: e.item.component.placeholder || `请输入${e.item.label}`
      }, e.item.component), null, 16, ["modelValue", "style", "placeholder"])) : p("", !0),
      e.item.component.name == "el-select" ? (m(), i(n(W), s({
        key: 1,
        modelValue: e.data[e.item.prop],
        "onUpdate:modelValue": r[1] || (r[1] = (l) => e.data[e.item.prop] = l),
        clearable: "",
        style: { width: n(u)(e.item.component.width) },
        placeholder: e.item.component.placeholder || `请选择${e.item.label}`
      }, e.item.component), {
        default: h(() => [
          (m(!0), f(b, null, S(e.item.component.options, (l) => (m(), i(n(j), {
            key: l.value,
            label: l.label || l,
            value: l.value || l.label || l
          }, {
            default: h(() => [
              e.item.component.showValue ? (m(), f("div", G, [
                w("span", null, V(l.label), 1),
                w("span", H, V(l.value), 1)
              ])) : p("", !0)
            ]),
            _: 2
          }, 1032, ["label", "value"]))), 128))
        ]),
        _: 1
      }, 16, ["modelValue", "style", "placeholder"])) : p("", !0),
      e.item.component.name == "el-date" ? (m(), i(n(q), s({
        key: 2,
        modelValue: e.data[e.item.prop],
        "onUpdate:modelValue": r[2] || (r[2] = (l) => e.data[e.item.prop] = l),
        type: "date",
        "value-format": "YYYY-MM-DD",
        shortcuts: y,
        style: { width: n(u)(e.item.component.width) },
        placeholder: e.item.component.placeholder || `请选择${e.item.label}`
      }, e.item.component), null, 16, ["modelValue", "style", "placeholder"])) : p("", !0),
      e.item.component.name == "u-search2" ? (m(), i(n(A), s({
        key: 3,
        modelValue: e.data[e.item.prop],
        "onUpdate:modelValue": r[3] || (r[3] = (l) => e.data[e.item.prop] = l),
        onRemoteSearch: e.item.component.remoteSearch,
        style: { width: n(u)(e.item.component.width) },
        placeholder: e.item.component.placeholder || `请输入${e.item.label}`
      }, e.item.component), null, 16, ["modelValue", "onRemoteSearch", "style", "placeholder"])) : p("", !0)
    ], 64)) : e.item.prop ? v(e.$slots, `form-${e.item.prop}`, {
      key: 1,
      item: e.item,
      data: e.data
    }, void 0, !0) : p("", !0);
  }
});
const K = /* @__PURE__ */ E(J, [["__scopeId", "data-v-ca5a3471"]]), Q = { class: "u-form" }, X = /* @__PURE__ */ $({
  name: "UForm",
  __name: "form",
  props: {
    form: {},
    group: {},
    validate: { type: Function },
    inline: { type: Boolean }
  },
  setup(c, { expose: y }) {
    const e = c, r = D();
    function l() {
      let t = e.form;
      L(t, {
        labelWidth: 80,
        labelPosition: "right"
      }), t.items && t.items.forEach((a) => {
        let o = a.component;
        o && o.name == "el-select" && o.remoteSearch && o.remoteSearch((d) => o.options = d);
      }), t.data = t.items.reduce((a, o) => (a[o.prop] || (a[o.prop] = o.value), a), t.data);
    }
    l();
    function F(t) {
      let a = e.form.group;
      return a && a.type == "tabs" ? a.value == t : a != null && a.type ? e.group == t : !0;
    }
    function U(t) {
      var a;
      (a = r.value) == null || a.validate((o, d) => {
        t(o, d);
      });
    }
    return y({
      validate: U,
      resetFields: () => {
        var t;
        return (t = r.value) == null ? void 0 : t.resetFields();
      }
    }), (t, a) => (m(), f("div", Q, [
      g(n(M), s({
        ref_key: "formRef",
        ref: r,
        model: t.form.data,
        "status-icon": "",
        "label-width": t.form.labelWidth,
        "label-position": t.form.labelPosition
      }, t.$props), {
        default: h(() => [
          (m(!0), f(b, null, S(t.form.items, (o, d) => I((m(), i(n(O), {
            key: d,
            label: o.label,
            prop: o.prop,
            class: R(`form-${o.prop}`),
            rules: o.required ? { required: !0, message: `${o.label}不能为空`, trigger: "change" } : o.rule,
            "label-width": o.labelWidth,
            style: z({ width: n(u)(o.width) })
          }, {
            default: h(() => [
              g(K, {
                item: o,
                data: t.form.data
              }, B({ _: 2 }, [
                o.component ? void 0 : {
                  name: `form-${o.prop}`,
                  fn: h((P) => [
                    v(t.$slots, `form-${o.prop}`, C(Y(P)), void 0, !0)
                  ]),
                  key: "0"
                }
              ]), 1032, ["item", "data"])
            ]),
            _: 2
          }, 1032, ["label", "prop", "class", "rules", "label-width", "style"])), [
            [N, F(o.group)]
          ])), 128)),
          v(t.$slots, "default", {}, void 0, !0)
        ]),
        _: 3
      }, 16, ["model", "label-width", "label-position"])
    ]));
  }
});
const Z = /* @__PURE__ */ E(X, [["__scopeId", "data-v-b75a2554"]]), re = Z;
export {
  re as U
};
