// undraw-ui/es/components/dialog/index.js
import { defineComponent as h, ref as r, watch as i, openBlock as a, createBlock as V, unref as u, createSlots as g, withCtx as d, createElementVNode as c, createElementBlock as p, createVNode as w, renderSlot as v, pushScopeId as k, popScopeId as y } from "vue";
import { ElDialog as C, ElScrollbar as b } from "element-plus";
import { _ as z } from "../anchor/index.js";
/*! UndrawUi v1.2.6 */
const f = (o) => (k("data-v-e4946d12"), o = o(), y(), o), B = {
  key: 0,
  width: "16",
  height: "16",
  viewBox: "0 0 1024 1024",
  version: "1.1",
  xmlns: "http://www.w3.org/2000/svg"
}, M = /* @__PURE__ */ f(() => /* @__PURE__ */ c("path", {
  d: "M704 354.133333H896c17.066667 0 34.133333-12.8 34.133333-34.133333s-17.066667-29.866667-34.133333-29.866667h-162.133333V128c0-17.066667-12.8-34.133333-34.133334-34.133333s-25.6 17.066667-25.6 34.133333v192c0 17.066667 12.8 34.133333 29.866667 34.133333zM320 98.133333c-17.066667 0-29.866667 12.8-29.866667 29.866667v162.133333H128c-17.066667 0-34.133333 12.8-34.133333 34.133334s17.066667 29.866667 34.133333 29.866666h192c17.066667 0 34.133333-12.8 34.133333-34.133333V128c0-17.066667-17.066667-29.866667-34.133333-29.866667zM320 674.133333H128c-17.066667 0-34.133333 12.8-34.133333 34.133334s12.8 34.133333 34.133333 34.133333h162.133333V896c0 17.066667 12.8 34.133333 34.133334 34.133333s34.133333-12.8 34.133333-34.133333v-192c-4.266667-17.066667-21.333333-29.866667-38.4-29.866667zM896 674.133333h-192c-17.066667 0-34.133333 12.8-34.133333 34.133334V896c0 17.066667 12.8 34.133333 34.133333 34.133333s34.133333-12.8 34.133333-34.133333v-162.133333H896c17.066667 0 34.133333-12.8 34.133333-34.133334s-17.066667-25.6-34.133333-25.6z",
  "p-id": "5080",
  fill: "currentColor"
}, null, -1)), S = [
  M
], H = {
  key: 1,
  width: "16",
  height: "16",
  viewBox: "0 0 1024 1024",
  xmlns: "http://www.w3.org/2000/svg"
}, E = /* @__PURE__ */ f(() => /* @__PURE__ */ c("path", {
  fill: "currentColor",
  d: "m160 96.064 192 .192a32 32 0 0 1 0 64l-192-.192V352a32 32 0 0 1-64 0V96h64v.064zm0 831.872V928H96V672a32 32 0 1 1 64 0v191.936l192-.192a32 32 0 1 1 0 64l-192 .192zM864 96.064V96h64v256a32 32 0 1 1-64 0V160.064l-192 .192a32 32 0 1 1 0-64l192-.192zm0 831.872-192-.192a32 32 0 0 1 0-64l192 .192V672a32 32 0 1 1 64 0v256h-64v-.064z"
}, null, -1)), I = [
  E
], U = /* @__PURE__ */ h({
  name: "UDialog",
  __name: "dialog",
  props: {
    title: {},
    modelValue: { type: Boolean },
    width: {},
    center: { type: Boolean },
    top: {},
    beforeClose: {},
    closeOnClickModal: { type: Boolean, default: !0 }
  },
  emits: ["update:modelValue"],
  setup(o, { emit: m }) {
    const _ = o, l = r(!1), t = r(!1);
    return i(
      () => _.modelValue,
      (e) => {
        l.value = e;
      },
      {
        immediate: !0
      }
    ), i(
      () => l.value,
      (e) => {
        m("update:modelValue", e);
      }
    ), (e, s) => (a(), V(u(C), {
      modelValue: l.value,
      "onUpdate:modelValue": s[1] || (s[1] = (n) => l.value = n),
      "close-on-click-modal": e.closeOnClickModal,
      class: "u-dialog",
      title: e.title,
      width: e.width,
      top: e.top,
      fullscreen: t.value,
      center: e.center,
      "before-close": e.beforeClose,
      draggable: ""
    }, g({
      default: d(() => [
        c("div", {
          class: "full-screen",
          onClick: s[0] || (s[0] = (n) => t.value = !t.value)
        }, [
          t.value ? (a(), p("svg", B, S)) : (a(), p("svg", H, I))
        ]),
        w(u(b), { ref: "scrollbarRef" }, {
          default: d(() => [
            v(e.$slots, "default", {}, void 0, !0)
          ]),
          _: 3
        }, 512)
      ]),
      _: 2
    }, [
      e.$slots.footer ? {
        name: "footer",
        fn: d(() => [
          v(e.$slots, "footer", {}, void 0, !0)
        ]),
        key: "0"
      } : void 0
    ]), 1032, ["modelValue", "close-on-click-modal", "title", "width", "top", "fullscreen", "center", "before-close"]));
  }
});
const $ = /* @__PURE__ */ z(U, [["__scopeId", "data-v-e4946d12"]]), x = $;
export {
  x as U
};
