// undraw-ui/es/components/editor/index.js
import { defineComponent as Z, computed as E, ref as x, watch as q, nextTick as G, onMounted as ue, onUnmounted as de, openBlock as L, createElementBlock as C, createVNode as V, unref as T, withCtx as fe, Fragment as J, renderList as Q, normalizeClass as W, renderSlot as X, createElementVNode as w, createCommentVNode as ve, toDisplayString as pe, withDirectives as D, vShow as Y, useCssVars as me, toRefs as ge, reactive as K, withKeys as j, withModifiers as z, mergeProps as he, pushScopeId as we, popScopeId as _e } from "vue";
import { cloneDeep as Se, isEmpty as ye, UToast as xe } from "undraw-ui";
import { ElScrollbar as Le, ElEmpty as Ce, ClickOutside as Me } from "element-plus";
import { _ as ee } from "../anchor/index.js";
/*! UndrawUi v1.2.6 */
const Ne = ["onClick"], Re = { class: "userInfo" }, ke = ["src"], be = { class: "username" }, Ee = { class: "empty" }, Te = /* @__PURE__ */ Z({
  __name: "mention",
  props: {
    target: {},
    data: {},
    alias: {},
    showAvatar: { type: Boolean },
    showMention: { type: Boolean }
  },
  emits: ["select"],
  setup(S, { expose: A, emit: u }) {
    const i = S, _ = E(() => {
      var o;
      return (o = i.target) == null ? void 0 : o.value;
    }), l = x(), r = x(), s = x(0);
    q(
      () => i.showMention,
      (o) => {
        o && (s.value = 0, G(() => {
          r.value && r.value.setScrollTop(0);
        }));
      }
    );
    function d() {
      var p, h;
      let o = (p = M()) == null ? void 0 : p.getBoundingClientRect(), g = (h = _.value) == null ? void 0 : h.getBoundingClientRect(), f = l.value;
      f && o && g && (f.style.top = `${o.top - g.top + 20}px`, f.style.left = `${o.left - g.left}px`);
    }
    function M() {
      var o;
      try {
        return (o = window.getSelection()) == null ? void 0 : o.getRangeAt(0);
      } catch {
        console.log("光标丢失了!");
        return;
      }
    }
    const k = (o) => {
      var g;
      if (s.value += o, i.data && (s.value < 0 ? s.value = i.data.length - 1 : s.value >= i.data.length && (s.value = 0)), r.value) {
        const f = r.value.wrapRef.children[0].children[s.value];
        if (f) {
          const p = r.value.wrapRef.offsetHeight || 0;
          (g = r.value) == null || g.setScrollTop((s.value - p / f.offsetHeight + 1) * f.offsetHeight);
        }
      }
    }, H = (o) => {
      s.value = o, i.data && u("select", i.data[s.value]);
    };
    function I() {
      return i.data && i.data[s.value];
    }
    function N(o) {
      d();
    }
    return ue(() => {
      var o;
      (o = _.value) == null || o.addEventListener("input", N);
    }), de(() => {
      var o;
      (o = _.value) == null || o.removeEventListener("input", N);
    }), A({
      moveSelection: k,
      getSelectItem: I
    }), (o, g) => (L(), C("ul", {
      ref_key: "mentionRef",
      ref: l,
      class: "mention-list"
    }, [
      V(T(Le), {
        ref_key: "scrollbarRef",
        ref: r
      }, {
        default: fe(() => [
          (L(!0), C(J, null, Q(o.data, (f, p) => (L(), C("li", {
            key: p,
            class: W({ hover: p === s.value }),
            onClick: (h) => H(p)
          }, [
            X(o.$slots, "mention-user", {
              item: f,
              index: p
            }, () => [
              w("div", Re, [
                o.showAvatar ? (L(), C("img", {
                  key: 0,
                  src: f[o.alias.avatar],
                  width: "30",
                  class: "avatar"
                }, null, 8, ke)) : ve("", !0),
                w("span", be, pe(f[o.alias.username]), 1)
              ])
            ], !0)
          ], 10, Ne))), 128)),
          D(w("div", Ee, [
            V(T(Ce), { description: "暂无匹配数据" })
          ], 512), [
            [Y, !(o.data && o.data.length)]
          ])
        ]),
        _: 3
      }, 512)
    ], 512));
  }
});
const Ae = /* @__PURE__ */ ee(Te, [["__scopeId", "data-v-4f3160d0"]]), He = (S) => (we("data-v-7653d7e4"), S = S(), _e(), S), Ie = ["placeholder", "innerHTML"], $e = {
  ref: "imageRef",
  class: "image-preview-box"
}, Ve = ["src"], De = ["onClick"], Be = /* @__PURE__ */ He(() => /* @__PURE__ */ w("svg", {
  "data-v-48a7e3c5": "",
  "data-v-7c7c7498": "",
  width: "12",
  height: "12",
  viewBox: "0 0 12 12",
  fill: "none",
  xmlns: "http://www.w3.org/2000/svg"
}, [
  /* @__PURE__ */ w("rect", {
    width: "12",
    height: "12",
    rx: "2",
    fill: "#86909C"
  }),
  /* @__PURE__ */ w("path", {
    "data-v-48a7e3c5": "",
    "data-v-7c7c7498": "",
    "fill-rule": "evenodd",
    "clip-rule": "evenodd",
    d: "M5.98095 5.49307L8.22012 3.25389C8.28521 3.18881 8.39074 3.18881 8.45582 3.25389L8.69153 3.4896C8.75661 3.55468 8.75661 3.66021 8.69153 3.7253L6.45235 5.96447L8.69153 8.20364C8.75661 8.26873 8.75661 8.37426 8.69153 8.43934L8.45582 8.67505C8.39074 8.74013 8.28521 8.74013 8.22012 8.67505L5.98095 6.43587L3.74178 8.67505C3.67669 8.74013 3.57116 8.74013 3.50608 8.67505L3.27037 8.43934C3.20529 8.37426 3.20529 8.26873 3.27037 8.20364L5.50954 5.96447L3.27037 3.7253C3.20529 3.66021 3.20529 3.55468 3.27037 3.4896L3.50608 3.25389C3.57116 3.18881 3.67669 3.18881 3.74178 3.25389L5.98095 5.49307Z",
    fill: "white"
  })
], -1)), Pe = [
  Be
], Ue = /* @__PURE__ */ Z({
  name: "UEditor",
  __name: "editor",
  props: {
    placeholder: {},
    modelValue: {},
    minHeight: { default: 30 },
    mention: {},
    imgList: {}
  },
  emits: ["update:modelValue", "input", "focus", "blur", "submit", "mention-search", "paste", "change-img-list"],
  setup(S, { expose: A, emit: u }) {
    var O;
    const i = S;
    me((e) => ({
      "0977f884": H.value,
      b90f9588: I.value
    }));
    const { imgList: _ } = ge(i), l = K({
      active: !1,
      isLocked: !1,
      showMention: !1
    }), r = K({
      focusNode: (O = window.getSelection()) == null ? void 0 : O.focusNode,
      index: -1,
      searchStr: ""
    }), s = x(), d = x(), M = x(), k = x(), H = E(() => i.minHeight + "px"), I = E(() => i.minHeight == 30 ? "4px 10px" : "8px 12px"), N = E(() => {
      let e = Se(i.mention || {});
      return e.showMention = !1, e.target = d, e.alias = o(e == null ? void 0 : e.alias, {
        id: "id",
        username: "username",
        avatar: "avatar"
      }), e;
    });
    function o(e, t) {
      return Object.assign({}, t, e);
    }
    q(
      () => i.modelValue,
      (e, t) => {
        e == "<br>" && B(), l.isLocked || (k.value = e);
      }
    );
    function g(e) {
      u("focus", e), l.isLocked = !0, l.active = !0;
    }
    function f() {
      G(() => {
        var e;
        (e = d.value) == null || e.focus(), s.value = h();
      });
    }
    function p(e) {
      var t;
      s.value = h(), u("blur", e), (t = d.value) != null && t.innerHTML || (l.active = !1), l.isLocked = !1;
    }
    function h(e) {
      var t;
      try {
        return e && e.focus(), (t = window.getSelection()) == null ? void 0 : t.getRangeAt(0);
      } catch {
        throw "没有选择任何文本";
      }
    }
    function B() {
      d.value && (d.value.innerHTML = "", u("update:modelValue", d.value.innerHTML), l.active = !1);
    }
    function te() {
      var n, a;
      let e = "";
      const t = window.getSelection();
      return (n = t == null ? void 0 : t.focusNode) != null && n.data && (r.focusNode = t.focusNode, e = (a = t.focusNode) == null ? void 0 : a.data.slice(0, t.focusOffset)), e;
    }
    function $() {
      let e = null, t = te();
      const n = t == null ? void 0 : t.lastIndexOf("@");
      return n !== -1 && (e = t.slice(n + 1), r.index = n, r.searchStr = e), e;
    }
    function P() {
      let e = $();
      return i.mention != null && e != null && !(e.includes(" ") || e.trim() != e);
    }
    function ne(e) {
      if (s.value = h(), l.showMention = P(), l.showMention) {
        let n = $();
        u("mention-search", n || "");
      }
      const { innerHTML: t } = e.target;
      u("update:modelValue", t), u("input", e);
    }
    function oe() {
      s.value = h(), $(), l.showMention = P();
    }
    const ae = (e) => {
      var t, n, a;
      if (e.ctrlKey && e.key == "Enter")
        l.showMention = !1, ye(i.modelValue.replace(/&nbsp;|<br>| /g, "")) ? xe({ message: "内容不能为空", type: "info" }) : u("submit");
      else if (e.key == "Enter" && l.showMention)
        e.preventDefault(), U((t = M.value) == null ? void 0 : t.getSelectItem());
      else if (e.key != "Enter") {
        if (e.key == "Backspace") {
          let c = window.getSelection(), m = s.value, v = (n = c == null ? void 0 : c.focusNode) == null ? void 0 : n.parentNode;
          v && m && c && v.tagName == "SPAN" && v.classList.contains("at-span") && (m.setStartBefore(v), m.setEndAfter(((a = v.nextSibling) == null ? void 0 : a.textContent) == " " ? v.nextSibling : v), m.deleteContents(), c.removeAllRanges(), c.addRange(m), l.showMention = !1, e.preventDefault());
        }
      }
    };
    function se(e) {
      const t = e.clipboardData;
      if (t) {
        const n = t.getData("text/plain"), a = t.items.length > 0 ? t.items[0].getAsFile() : null;
        n ? (e.preventDefault(), document.execCommand("insertText", !1, n)) : a && (e.preventDefault(), u("paste", e, a));
      }
    }
    const le = (e, t, n = "var(--u-color-primary)") => {
      const a = document.createElement("span");
      return a.className = "at-span", a.style.color = n, a.id = e.toString(), a.innerHTML = `@${t}`, a;
    };
    function ie(e) {
      var v, R;
      let t = d.value;
      s.value || (s.value = h(t));
      let n = s.value, a = getSelection();
      if (s.value.deleteContents(), t && a) {
        if (a.removeAllRanges(), a.addRange(s.value), ((v = a.anchorNode) == null ? void 0 : v.nodeName) != "#text") {
          var c = document.createTextNode(e);
          n.insertNode(c), a.removeAllRanges(), a.addRange(n), a.collapseToEnd();
        } else {
          let y = n.startContainer;
          var m = n.startOffset;
          y.insertData(m, e), n.setStart(y, m + e.length), n.collapse(!0), a.removeAllRanges(), a.addRange(n);
        }
        s.value = h(t), u("update:modelValue", ((R = d.value) == null ? void 0 : R.innerHTML) || "");
        const b = d.value;
        u("input", b);
      }
    }
    const re = (e) => {
      var t;
      (t = _ == null ? void 0 : _.value) == null || t.splice(e, 1);
    };
    function U(e) {
      var m, v, R;
      const t = window.getSelection();
      let n = s.value, a = N.value.alias, c = (m = r.focusNode) == null ? void 0 : m.parentNode;
      if (e && t && n) {
        c && c.tagName == "SPAN" && c.classList.contains("at-span") ? (n.setStartBefore(c), n.setEndAfter(((v = c.nextSibling) == null ? void 0 : v.textContent) == " " ? c.nextSibling : c), n.deleteContents(), n.insertNode(document.createTextNode("@"))) : (n.setStart(r.focusNode, r.index), n.setEnd(r.focusNode, r.index + 1 + r.searchStr.length)), n.deleteContents();
        const b = le(e[a.id], e[a.username]);
        n.insertNode(b);
        const y = document.createElement("span");
        y.innerHTML = " ", y.style.whiteSpace = "pre-wrap", b.after(y), n.setStartAfter(y), t.removeAllRanges(), t.addRange(n), l.showMention = !1, u("update:modelValue", ((R = d.value) == null ? void 0 : R.innerHTML) || "");
        const ce = d.value;
        u("input", ce);
      }
      u("mention-search", "");
    }
    function F(e) {
      var t;
      (t = M.value) == null || t.moveSelection(e);
    }
    return A({
      addText: ie,
      clear: B,
      focus: f
    }), (e, t) => D((L(), C("div", {
      class: W(["u-editor", { active: l.active }])
    }, [
      w("div", {
        ref_key: "editorRef",
        ref: d,
        class: "rich-input",
        contenteditable: "true",
        placeholder: e.placeholder,
        onFocus: g,
        onInput: ne,
        onBlur: p,
        onKeydown: [
          ae,
          t[0] || (t[0] = j(z((n) => F(-1), ["prevent"]), ["up"])),
          t[1] || (t[1] = j(z((n) => F(1), ["prevent"]), ["down"]))
        ],
        onPaste: se,
        onMouseup: oe,
        innerHTML: k.value
      }, null, 40, Ie),
      w("div", $e, [
        (L(!0), C(J, null, Q(T(_), (n, a) => (L(), C("div", {
          key: a,
          class: "image-preview"
        }, [
          w("img", {
            src: n,
            alt: ""
          }, null, 8, Ve),
          w("div", {
            class: "clean-btn",
            onClick: (c) => re(a)
          }, Pe, 8, De)
        ]))), 128))
      ], 512),
      X(e.$slots, "footer", {}, void 0, !0),
      D(V(Ae, he({
        ref_key: "mentionRef",
        ref: M
      }, N.value, {
        "show-mention": l.showMention,
        onSelect: U
      }), null, 16, ["show-mention"]), [
        [Y, l.showMention]
      ])
    ], 2)), [
      [T(Me), () => l.showMention = !1, M.value]
    ]);
  }
});
const Fe = /* @__PURE__ */ ee(Ue, [["__scopeId", "data-v-7653d7e4"]]), Ze = Fe;
export {
  Ze as U
};
