// undraw-ui/es/components/table/index.js
import { defineComponent as _, openBlock as i, createBlock as c, unref as n, createSlots as fe, withCtx as p, createVNode as w, normalizeClass as U, renderSlot as $, normalizeProps as H, mergeProps as z, createElementBlock as v, Fragment as E, renderList as A, createTextVNode as ge, toDisplayString as K, createCommentVNode as y, guardReactiveProps as x, pushScopeId as G, popScopeId as J, createElementVNode as h, ref as M, reactive as we, inject as N, computed as Y, watch as ve, onMounted as be, onUnmounted as ye, withDirectives as ke, normalizeStyle as Ce, withKeys as ze, nextTick as Ve } from "vue";
import { ElTableColumn as R, ElFormItem as Ee, ElImage as Se, ElInput as Q, ElSelect as $e, ElOption as Ie, ElDatePicker as Be, ElTag as Me, ElForm as Ue, ElTable as He, ElEmpty as Ae, ElLink as S, vLoading as Re, ElPagination as Le, ElMessageBox as Te } from "element-plus";
import { UIcon as V, USearch2 as De, debounce as F, cloneDeep as Pe } from "undraw-ui";
import { t as je, m as We } from "../../util/basic.js";
import "../../util/clean-html.js";
import { T as O } from "../../util/datetime.js";
import { _ as X } from "../anchor/index.js";
/*! UndrawUi v1.2.6 */
const qe = (g) => (G("data-v-3d3d8d61"), g = g(), J(), g), Ne = /* @__PURE__ */ qe(() => /* @__PURE__ */ h("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 1024 1024",
  "data-v-974b4c7f": ""
}, [
  /* @__PURE__ */ h("path", {
    fill: "currentColor",
    d: "M160 160v704h704V160zm-32-64h768a32 32 0 0 1 32 32v768a32 32 0 0 1-32 32H128a32 32 0 0 1-32-32V128a32 32 0 0 1 32-32"
  }),
  /* @__PURE__ */ h("path", {
    fill: "currentColor",
    d: "M384 288q64 0 64 64t-64 64q-64 0-64-64t64-64M185.408 876.992l-50.816-38.912L350.72 556.032a96 96 0 0 1 134.592-17.856l1.856 1.472 122.88 99.136a32 32 0 0 0 44.992-4.864l216-269.888 49.92 39.936-215.808 269.824-.256.32a96 96 0 0 1-135.04 14.464l-122.88-99.072-.64-.512a32 32 0 0 0-44.8 5.952z"
  })
], -1)), Ye = /* @__PURE__ */ _({
  __name: "item",
  props: {
    item: {},
    noAdd: { type: Boolean },
    sort: {}
  },
  setup(g) {
    const I = g, C = [
      {
        text: "今天",
        value: /* @__PURE__ */ new Date()
      },
      {
        text: "昨天",
        value: new O().add(-1, "day")
      },
      {
        text: "一周前",
        value: new O().add(-1, "week")
      }
    ];
    function m(e, o) {
      return e || (e = [{ label: "", type: "primary" }]), [e.find((s) => s.value == o || s.label == o) || { label: "", type: "primary" }];
    }
    return (e, o) => (i(), c(n(R), {
      "show-overflow-tooltip": e.item.overflow,
      label: e.item.label,
      prop: e.item.prop,
      width: e.item.width,
      "min-width": e.item.minWidth,
      sortable: e.item.sort !== void 0 && (I.sort == "custom" ? "custom" : !0),
      align: e.item.align,
      type: e.item.type,
      fixed: e.item.fixed
    }, fe({ _: 2 }, [
      ["selection", "index"].includes(e.item.type || "") ? void 0 : {
        name: "default",
        fn: p((d) => [
          w(n(Ee), {
            prop: `${d.$index}.${e.item.prop}`,
            rules: e.item.required ? { required: !0, message: `${e.item.label}不能为空`, trigger: "change" } : e.item.rule,
            class: U({ "no-add": e.noAdd, center: e.item.align == "center" })
          }, {
            default: p(() => [
              !e.item.component && (!e.item.type || e.item.type == "basic") ? $(e.$slots, "basic", H(z({ key: 0 }, d)), void 0, !0) : e.item.prop && e.item.type == "img" ? (i(), c(n(Se), {
                key: 1,
                src: d.row[e.item.prop],
                "preview-src-list": [d.row[e.item.prop]],
                "preview-teleported": "",
                fit: "fill",
                class: "w-10 h-10"
              }, {
                error: p(() => [
                  w(n(V), { style: { display: "flex", "justify-content": "center", "align-items": "center" } }, {
                    default: p(() => [
                      Ne
                    ]),
                    _: 1
                  })
                ]),
                _: 2
              }, 1032, ["src", "preview-src-list"])) : e.item.component ? (i(), v(E, { key: 2 }, [
                e.item.prop && e.item.component.name == "el-input" ? (i(), c(n(Q), z({
                  key: 0,
                  modelValue: d.row[e.item.prop],
                  "onUpdate:modelValue": (s) => d.row[e.item.prop] = s,
                  clearable: "",
                  placeholder: e.item.component.placeholder || `请选择${e.item.label}`
                }, e.item.component), null, 16, ["modelValue", "onUpdate:modelValue", "placeholder"])) : e.item.prop && e.item.component.name == "el-select" ? (i(), c(n($e), z({
                  key: 1,
                  modelValue: d.row[e.item.prop],
                  "onUpdate:modelValue": (s) => d.row[e.item.prop] = s,
                  clearable: "",
                  placeholder: e.item.component.placeholder || `请选择${e.item.label}`
                }, e.item.component), {
                  default: p(() => [
                    (i(!0), v(E, null, A(e.item.component.options, (s) => (i(), c(n(Ie), {
                      key: s.value,
                      label: s.label,
                      value: s.value || s.label || s
                    }, null, 8, ["label", "value"]))), 128))
                  ]),
                  _: 2
                }, 1040, ["modelValue", "onUpdate:modelValue", "placeholder"])) : e.item.prop && e.item.component.name == "el-date" ? (i(), c(n(Be), z({
                  key: 2,
                  modelValue: d.row[e.item.prop],
                  "onUpdate:modelValue": (s) => d.row[e.item.prop] = s,
                  type: "date",
                  "value-format": "YYYY-MM-DD",
                  shortcuts: C,
                  placeholder: e.item.component.placeholder || `请选择${e.item.label}`
                }, e.item.component), null, 16, ["modelValue", "onUpdate:modelValue", "placeholder"])) : e.item.prop && e.item.component.name == "el-tag" ? (i(!0), v(E, { key: 3 }, A(m(e.item.component.options, d.row[e.item.prop]), (s) => (i(), c(n(Me), {
                  type: s.type
                }, {
                  default: p(() => [
                    ge(K(s.label), 1)
                  ]),
                  _: 2
                }, 1032, ["type"]))), 256)) : e.item.prop && e.item.component.name == "u-search2" ? (i(), c(n(De), z({
                  key: 4,
                  modelValue: d.row[e.item.prop],
                  "onUpdate:modelValue": (s) => d.row[e.item.prop] = s,
                  onRemoteSearch: e.item.component.remoteSearch,
                  placeholder: e.item.component.placeholder || `请选择${e.item.label}`
                }, e.item.component), null, 16, ["modelValue", "onUpdate:modelValue", "onRemoteSearch", "placeholder"])) : y("", !0)
              ], 64)) : y("", !0),
              e.item.type == "custom" ? (i(), v("div", {
                key: 3,
                class: U({ overflow: e.item.overflow })
              }, [
                $(e.$slots, `table-${e.item.prop}`, H(x(d)), void 0, !0)
              ], 2)) : y("", !0)
            ]),
            _: 2
          }, 1032, ["prop", "rules", "class"])
        ]),
        key: "0"
      }
    ]), 1032, ["show-overflow-tooltip", "label", "prop", "width", "min-width", "sortable", "align", "type", "fixed"]));
  }
});
const Fe = /* @__PURE__ */ X(Ye, [["__scopeId", "data-v-3d3d8d61"]]), k = (g) => (G("data-v-ecf56028"), g = g(), J(), g), Oe = { class: "u-table" }, _e = /* @__PURE__ */ k(() => /* @__PURE__ */ h("svg", {
  "data-v-c4997ac3": "",
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 1024 1024"
}, [
  /* @__PURE__ */ h("path", {
    fill: "currentColor",
    d: "M480 480V128a32 32 0 0 1 64 0v352h352a32 32 0 1 1 0 64H544v352a32 32 0 1 1-64 0V544H128a32 32 0 0 1 0-64h352z"
  })
], -1)), Ke = /* @__PURE__ */ k(() => /* @__PURE__ */ h("svg", {
  "data-v-c4997ac3": "",
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 1024 1024"
}, [
  /* @__PURE__ */ h("path", {
    fill: "currentColor",
    d: "M128 544h768a32 32 0 1 0 0-64H128a32 32 0 0 0 0 64z"
  })
], -1)), xe = /* @__PURE__ */ k(() => /* @__PURE__ */ h("svg", {
  "data-v-c4997ac3": "",
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 1024 1024"
}, [
  /* @__PURE__ */ h("path", {
    fill: "currentColor",
    d: "m199.04 672.64 193.984 112 224-387.968-193.92-112-224 388.032zm-23.872 60.16 32.896 148.288 144.896-45.696L175.168 732.8zM455.04 229.248l193.92 112 56.704-98.112-193.984-112-56.64 98.112zM104.32 708.8l384-665.024 304.768 175.936L409.152 884.8h.064l-248.448 78.336L104.32 708.8zm384 254.272v-64h448v64h-448z"
  })
], -1)), Ge = /* @__PURE__ */ k(() => /* @__PURE__ */ h("span", null, "修改", -1)), Je = /* @__PURE__ */ k(() => /* @__PURE__ */ h("svg", {
  "data-v-c4997ac3": "",
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 1024 1024"
}, [
  /* @__PURE__ */ h("path", {
    fill: "currentColor",
    d: "M160 256H96a32 32 0 0 1 0-64h256V95.936a32 32 0 0 1 32-32h256a32 32 0 0 1 32 32V192h256a32 32 0 1 1 0 64h-64v672a32 32 0 0 1-32 32H192a32 32 0 0 1-32-32V256zm448-64v-64H416v64h192zM224 896h576V256H224v640zm192-128a32 32 0 0 1-32-32V416a32 32 0 0 1 64 0v320a32 32 0 0 1-32 32zm192 0a32 32 0 0 1-32-32V416a32 32 0 0 1 64 0v320a32 32 0 0 1-32 32z"
  })
], -1)), Qe = /* @__PURE__ */ k(() => /* @__PURE__ */ h("span", null, "删除", -1)), Xe = ["onDblclick"], Ze = `
        <path class="path" d="
          M 30 15
          L 28 17
          M 25.61 25.61
          A 15 15, 0, 0, 1, 15 30
          A 15 15, 0, 1, 1, 27.99 7.5
          L 15 15
        " style="stroke-width: 4px; fill: rgba(0, 0, 0, 0)"/>
      `, et = /* @__PURE__ */ _({
  name: "UTable",
  __name: "table",
  props: {
    table: {},
    maxHeight: {}
  },
  emits: ["onSelection", "inputBlur", "sortChange"],
  setup(g, { expose: I, emit: C }) {
    const m = g, e = M(), o = we({
      btnLoading: !1,
      loading: !1,
      // 表格加载动画
      visible: !1,
      current: 1,
      // 页数
      size: m.table.size || 10,
      // 页大小
      tableRowEditIndex: void 0,
      // 编辑的表格行
      tableColumnEditIndex: void 0,
      // 编辑的表格列
      noAdd: !1,
      sort: {},
      defaultSort: { prop: "", order: "ascending" }
    }), d = N("u-crud", {}), s = N("edit", () => {
    }), Z = Y(() => je(P.value).replace("px") - 40), ee = Y(() => m.table.total || m.table.data && m.table.data.length), te = (t) => {
      Te.confirm("你确定要删除吗？", "系统提示", {
        confirmButtonText: "确定",
        cancelButtonText: "取消",
        type: "warning",
        draggable: !0
      }).then(() => {
        d.remove && d.remove(t, () => {
          b();
        });
      });
    }, L = F(() => {
      o.loading = !0;
    }, 300), le = () => (L(), () => {
      setTimeout(() => {
        var t;
        L.cancel(), o.loading = !1, (t = e.value) == null || t.clearSelection();
      }, 100);
    });
    ve(
      () => m.table.total,
      (t) => {
        if (t) {
          let l = Math.ceil(t / o.size);
          o.current > l && (o.current = l, b());
        }
      }
    );
    const b = F(() => {
      let t = m.table.refresh;
      t && t(le(), o.current, o.size, o.sort);
    });
    function T(t) {
      o.current = t, b();
    }
    function oe(t) {
      let l = m.table.total;
      if (l) {
        let a = Math.ceil(l / t);
        o.current > a && T(a);
      }
      o.size = t, b();
    }
    function ae(t) {
      C("onSelection", t);
    }
    function ne(t, l) {
      if (l) {
        let a = function(u) {
          u.forEach((r) => {
            var q;
            const f = r.children;
            f && a(f), t.id != r.id && t.parentId == r.parentId && (r.children, (q = e.value) == null || q.toggleRowExpansion(r, !1));
          });
        };
        a(m.table.data);
      }
    }
    const ie = (t) => {
      o.tableRowEditIndex = t.$index, o.tableColumnEditIndex = t.column.id, Ve(() => {
        let l = document.querySelector("#u-table-input");
        l && (l.focus(), l.setSelectionRange && l.setSelectionRange(0, l.value.length));
      });
    }, re = (t) => {
      o.tableRowEditIndex = void 0, o.tableColumnEditIndex = void 0, C("inputBlur", t.row);
    };
    let B = () => m.table.columns.reduce((t, l) => (l.prop && (t[l.prop] = l.value), t), {});
    function D() {
      let t = m.table;
      We(t, {
        single: !0
      }), t.columns.forEach((l) => {
        l.minWidth || (l.minWidth = t.minWidth);
      }), o.sort = t.columns.reduce((l, a) => {
        if (a.sort !== void 0 && a.sort != null) {
          l[a.prop] = a.sort;
          let u = a.sort;
          switch (u) {
            case "asc":
              u = "ascending";
              break;
            case "desc":
              u = "descending";
              break;
            default:
              u = null;
          }
          o.defaultSort = { prop: a.prop || "", order: u };
        }
        return l;
      }, {}), o.noAdd = !t.columns.some((l) => l.type == "row-add"), o.noAdd || m.table.data.push(B()), b();
    }
    D();
    function de(t) {
      m.table.data.splice(t + 1, 0, B());
    }
    function se(t) {
      let l = m.table.data;
      l.splice(t, 1), (l == null || l && l.length < 1) && l.push(B());
    }
    function pe() {
      let t = m.table.data;
      t.length = 0, D();
    }
    function ue({ column: t }) {
      let l = { background: "#f5f7fa", color: "#606266", textAlign: "" };
      return m.table.columns.forEach((u) => {
        t.property == u.prop && (u.align == "right" && (l.textAlign = "left"), u.headerAlign && (l.textAlign = u.headerAlign));
      }), l;
    }
    function me({ column: t, prop: l, order: a }) {
      let u = o.sort;
      u && Object.keys(u).forEach((r) => {
        u[r] = null;
      }), a == "ascending" ? o.sort[l] = "asc" : a == "descending" ? o.sort[l] = "desc" : o.sort[l] = null, C("sortChange", o.sort);
    }
    const P = M(m.maxHeight), ce = () => {
      P.value = window.innerHeight - 300;
    };
    function j() {
      ce();
    }
    be(() => {
      j();
    }), ye(() => {
      window.removeEventListener("resize", j);
    });
    const W = M();
    function he(t) {
      var l;
      (l = W.value) == null || l.validate((a, u) => {
        t(a, u);
      });
    }
    return I({
      reload: b,
      validate: he,
      clear: pe
    }), (t, l) => (i(), v("div", Oe, [
      w(n(Ue), {
        ref_key: "formRef",
        ref: W,
        model: t.table.data
      }, {
        default: p(() => [
          ke((i(), c(n(He), {
            ref_key: "tableRef",
            ref: e,
            data: t.table.data,
            "max-height": t.maxHeight,
            stripe: "",
            border: "",
            "header-cell-style": ue,
            "row-key": "id",
            "element-loading-spinner": Ze,
            "element-loading-svg-view-box": "-10, -10, 50, 50",
            "element-loading-text": "拼命加载中",
            onExpandChange: ne,
            onSelectionChange: ae,
            onSortChange: me,
            "default-sort": o.defaultSort
          }, {
            empty: p(() => [
              w(n(Ae), {
                style: Ce({ height: Z.value + "px" }),
                "image-size": 200
              }, null, 8, ["style"])
            ]),
            default: p(() => [
              (i(!0), v(E, null, A(t.table.columns, (a, u) => (i(), v(E, { key: u }, [
                a.type == "row-add" ? (i(), c(n(R), {
                  key: 0,
                  label: a.label,
                  width: a.width,
                  "min-width": a.minWidth,
                  align: a.align
                }, {
                  default: p((r) => [
                    w(n(S), {
                      underline: !1,
                      type: "primary",
                      link: "",
                      onClick: (f) => de(r.$index)
                    }, {
                      default: p(() => [
                        w(n(V), {
                          size: "12px",
                          style: { "margin-right": "10px" }
                        }, {
                          default: p(() => [
                            _e
                          ]),
                          _: 1
                        })
                      ]),
                      _: 2
                    }, 1032, ["onClick"]),
                    w(n(S), {
                      underline: !1,
                      link: "",
                      onClick: (f) => se(r.$index)
                    }, {
                      default: p(() => [
                        w(n(V), { size: "12px" }, {
                          default: p(() => [
                            Ke
                          ]),
                          _: 1
                        })
                      ]),
                      _: 2
                    }, 1032, ["onClick"])
                  ]),
                  _: 2
                }, 1032, ["label", "width", "min-width", "align"])) : a.type == "operation" ? (i(), c(n(R), {
                  key: 1,
                  fixed: "right",
                  align: a.align,
                  label: "操作",
                  width: a.width
                }, {
                  default: p((r) => [
                    n(d) && n(d).update ? (i(), c(n(S), {
                      key: 0,
                      underline: !1,
                      type: "primary",
                      style: { "font-size": "12px", "margin-right": "5px" },
                      link: "",
                      onClick: (f) => n(s)(n(Pe)(r.row))
                    }, {
                      default: p(() => [
                        w(n(V), { size: "12px" }, {
                          default: p(() => [
                            xe
                          ]),
                          _: 1
                        }),
                        Ge
                      ]),
                      _: 2
                    }, 1032, ["onClick"])) : y("", !0),
                    n(d) && n(d).remove ? (i(), c(n(S), {
                      key: 1,
                      underline: !1,
                      type: "primary",
                      loading: o.btnLoading,
                      style: { "font-size": "12px", margin: "0" },
                      link: "",
                      onClick: (f) => te([r.row])
                    }, {
                      default: p(() => [
                        w(n(V), { size: "12px" }, {
                          default: p(() => [
                            Je
                          ]),
                          _: 1
                        }),
                        Qe
                      ]),
                      _: 2
                    }, 1032, ["loading", "onClick"])) : y("", !0)
                  ]),
                  _: 2
                }, 1032, ["align", "width"])) : (i(), c(Fe, {
                  key: 2,
                  item: a,
                  "no-add": o.noAdd,
                  sort: t.table.sort
                }, {
                  basic: p((r) => [
                    a.editor && o.tableRowEditIndex == r.$index && o.tableColumnEditIndex == r.column.id ? (i(), c(n(Q), {
                      key: 0,
                      id: "u-table-input",
                      modelValue: r.row[a.prop || ""],
                      "onUpdate:modelValue": (f) => r.row[a.prop || ""] = f,
                      onKeyup: l[0] || (l[0] = ze((f) => f.target.blur(), ["enter"])),
                      onBlur: (f) => re(r)
                    }, null, 8, ["modelValue", "onUpdate:modelValue", "onBlur"])) : (i(), v("div", {
                      key: 1,
                      class: U(["row-content", { overflow: a.overflow }]),
                      style: { width: "100%" },
                      onDblclick: (f) => ie(r)
                    }, K(r.row[a.prop || ""]), 43, Xe))
                  ]),
                  [`table-${a.prop}`]: p((r) => [
                    $(t.$slots, `table-${a.prop}`, H(x(r)), void 0, !0)
                  ]),
                  _: 2
                }, 1032, ["item", "no-add", "sort"]))
              ], 64))), 128)),
              $(t.$slots, "default", {}, void 0, !0)
            ]),
            _: 3
          }, 8, ["data", "max-height", "default-sort"])), [
            [n(Re), o.loading]
          ])
        ]),
        _: 3
      }, 8, ["model"]),
      t.table.type == "page" ? (i(), c(n(Le), {
        key: 0,
        "current-page": o.current,
        "onUpdate:currentPage": l[1] || (l[1] = (a) => o.current = a),
        layout: "total, sizes, prev, pager, next, jumper",
        small: "",
        "hide-on-single-page": t.table.single,
        total: ee.value,
        "page-sizes": [5, 10, 20, 40],
        "page-size": o.size,
        background: "",
        class: "flex flex-wrap",
        onCurrentChange: T,
        onSizeChange: oe
      }, null, 8, ["current-page", "hide-on-single-page", "total", "page-size"])) : y("", !0)
    ]));
  }
});
const tt = /* @__PURE__ */ X(et, [["__scopeId", "data-v-ecf56028"]]), st = tt;
export {
  st as U
};
